distributed/nanny.py

```diff
@@ -329,25 +329,37 @@ def start(self):
             yield self.running.wait()
             return
 
-        self.init_result_q = mp_context.Queue()
-        self.child_stop_q = mp_context.Queue()
-        self.process = AsyncProcess(
-            target=self._run,
-            kwargs=dict(worker_args=self.worker_args,
-                        worker_kwargs=self.worker_kwargs,
-                        worker_start_args=self.worker_start_args,
-                        silence_logs=self.silence_logs,
-                        init_result_q=self.init_result_q,
-                        child_stop_q=self.child_stop_q),
-        )
-        self.process.daemon = True
-        self.process.set_exit_callback(self._on_exit)
-        self.running = Event()
-        self.stopped = Event()
-        self.status = 'starting'
-        yield self.process.start()
-        if self.status == 'starting':
-            yield self._wait_until_running()
+        while True:
+            # FIXME: this sometimes stalls in _wait_until_running
+            # our temporary solution is to retry a few times if the process
+            # doesn't start up in five seconds
```

Comment:
@pitrou this was also interesting to find

This seems to resolve things pragmatically though.  No need for intervention, I just thought you might find it interesting.